common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -318,36 +317,36 @@ public boolean onTouch(final View view, final MotionEvent event) {
-                );
-                // We can only grant or deny permissions for video capture/camera.
-                // To avoid unintentionally granting requests for not defined permissions
-                // we check if the request is for camera.
-                if (!CameraPermissionRequest.isValidRequest(request)) {
-                    Logger.warn(methodTag, "Permission request is not for camera.");
-                    request.deny();
-                    return;
-                }
-                // There is a issue in ESTS UX where it sends multiple camera permission requests.
-                // So, if there is already a camera permission request in progress we handle it here.
-                if (mCameraPermissionRequest != null) {
-                    Logger.info(methodTag, "Repeated request, granted? " + mCameraPermissionRequest.isGranted());
-                    handleRepeatedCameraRequests(request);
-                    return;
-                }
-                Logger.info(methodTag, "New camera request.");
-                mCameraPermissionRequest = new CameraPermissionRequest(request);
-                // If the OS level permission was granted previously,
-                // we show the rationale to confirm the consent with the current user.
-                // Otherwise, show the system prompt.
-                if (isAppCameraPermissionGranted()) {
-                    Logger.info(methodTag, "Camera permission already granted.");
-                    showCameraRationale();
-                } else {
-                    requestCameraPermission();
-                }
+                requireActivity().runOnUiThread(() -> {
+                    // Log the permission request
+                    Logger.info(methodTag,
+                            "Permission requested from:" +request.getOrigin() +
+                                    " for resources:" + Arrays.toString(request.getResources())
+                    );
+                    // We can only grant or deny permissions for video capture/camera.
+                    // To avoid unintentionally granting requests for not defined permissions
+                    // we check if the request is for camera.
+                    if (!mCameraPermissionRequestHandler.setIfValid(request)) {
+                        Logger.warn(methodTag, "Permission request is not valid, returning.");
+                        return;
+                    }
+
+                    // If the OS level permission was granted previously,
+                    // we show the rationale to confirm the consent with the current user.
+                    // Otherwise, show the system prompt.
+                    if (isAppCameraPermissionGranted()) {
+                        Logger.info(methodTag, "App level camera permission already granted.");
+                        if(request.getOrigin() != null &&  "https://login.microsoftonline.com/".equalsIgnoreCase(request.getOrigin().toString())) {
```

Comment:
Nit: I think we can put "https://login.microsoftonline.com/" in a constant.

Reply:
sure